### ISSUE TITLE
User-defined named colors

### ISSUE THREAD

**Issue by ghochschild** [facelessuser/ColorHelper#42]
CSS color names are probably the most common and supported with `webcolors` but some languages have their own set of named colors. An option to add user-defined color names would be nice. R, for example, has lots of [named colors](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf).

The option could work like this: 

```
"colors_sets": {
     "R": {
          "aquamarine3": "#....",
          "seagreen4": "#....",
          ...
     },
     "SomeOtherSet": {
         "myred": "#fd0000"
    }
}
```

Now I have a named set of colors, which I could add to `allowed_colors` as part of the `color_scanning` option like this `"allowed_colors": ["R", "hex"]`

**Comment by facelessuser** [comment]
Notes:

This on the surface seems like a simple enough request, but the implications are bigger than it would seem.  Where there is one request like this, there will be others.

Originally this was a plugin that mainly did web colors, but already some features have slipped in that makes this more as  a general color plugin than just web specific.  This is reminiscent of BracketHighlighter in the early days.  BH at the time hardcoded a lot of its bracket specific rules.  This become more problematic as people wanted more and more language specific things handled, and I was forced to either personally handle them, or rewrite things to expose the knobs to the community so they could properly add all they wanted.

This request could be handled as a one shot, or this could be an opportunity to make ColorHelper more modular and treat each color type as a plugin.  This would allow me to document how a user can add their own colors and allow users to either pull request their special language colors or do it locally on their machine.  Currently, their is just too much a user needs to be aware of to add a feature like this themselves.

Though it puts more work on me upfront, taking initiative now will save me work later as I can push these kinds of things off on the community and focus on the core code only.

I need to think about this and see how I want to proceed.

**Comment by facelessuser** [comment]
I've decided on attempting to push color logic into modules.  I want to make it so I  can push these kind of requests off on the community.  This will probably in the next major release.  There are are just so many moving pieces like color translation, translation to other colors, UI elements, etc.  I need to make sure we can dynamically update this and not break the multiple threads we have running we dynamically as well.  But if all goes well, we should be able to have more flexible color options in the future.    We'll see how this all goes.

**Comment by facelessuser** [comment]
This will be included under the umbrella of the feature #141 

Closing as a duplicate.

@gir-bot add S: duplicate
